src/ascii_char.rs

```diff
@@ -299,54 +299,55 @@ impl AsciiChar {
     }
 
     /// Constructs an ASCII character from a `char` or `u8` without any checks.
+    #[inline]
     pub unsafe fn from_unchecked<C:ToAsciiChar>(ch: C) -> Self {
         ch.to_ascii_char_unchecked()
     }
 
     /// Converts an ASCII character into a `u8`.
     #[inline]
-    pub fn as_byte(&self) -> u8 {
-        *self as u8
+    pub fn as_byte(self) -> u8 {
```

Comment:
Technically this is a breaking change, but `self` always implements `Copy` so it just shifts the copying from the this function to the caller of `as_byte`. That means no code would really stop compiling, wouldn't it?